src/Tools/NodeDocumentationMarkdownGenerator/AssemblyHandler.cs

```diff
@@ -0,0 +1,387 @@
+                            functionGroup.AddFunctionDescriptor(descriptor);
+                        }
+                    }
+                    catch (Exception)
+                    {
+                        continue;
+                    }
+                }
+
+                if (functionGroups.Count > 0)
+                {
+                    AddZeroTouchNodesToSearch(functionGroups.Values.SelectMany(x => x.Values.Select(g => g)), ref nodeSearchModel);
+                }
+
+                fileInfos.AddRange(FileInfosFromSearchModel(nodeSearchModel));
+                Console.WriteLine($"{fileInfos.Count()} nodes found during scan");
+            }
+
+            return fileInfos;
+        }
+
+        private static List<FunctionDescriptor> GetFunctionDescriptorsFromDll(PathManager pathManager, Assembly asm)
+        {
+            var objectAsmName = typeof(object).Assembly.GetName().Name;
+            var descriptors = new List<FunctionDescriptor>();
+
+            string extension = System.IO.Path.GetExtension(asm.Location).ToLower();
+            if (extension != ".dll" && extension != ".exe")
+            {
+                return new List<FunctionDescriptor>();
+            }
+
+            // Getting ZT imports from CLRModuleTypes
+            DLLModule dllModule = null;
+            dllModule = DLLFFIHandler.GetModuleForInspection(asm);
+
+            dllModule.ScanModule(true);
+            List<CLRModuleType> moduleTypes = CLRModuleType.GetTypes((CLRModuleType mtype) => { return mtype.Module == dllModule; });
+
+            var customizationFile = LibraryCustomizationServices.GetForAssembly(asm.Location, pathManager);
+
+            foreach (var t in moduleTypes)
+            {
+                try
+                {
+                    var className = t.ClassNode.ClassName;
+                    var externalLib = t.ClassNode.ExternLibName;
+
+                    // For some reason mscorelib sometimes gets pass to here, so filtering it away.
+                    if (t.CLRType.Assembly.GetName().Name == objectAsmName ||
+                        t.ClassNode.ClassAttributes.HiddenInLibrary) continue;
+
+                    var ctorNodes = t.ClassNode.Procedures
+                        .OfType<ConstructorDefinitionNode>()
+                        .Where(c => c.Access == ProtoCore.CompilerDefinitions.AccessModifier.Public &&
+                                    c.MethodAttributes != null ?
+                                        !c.MethodAttributes.IsObsolete && !c.MethodAttributes.HiddenInLibrary :
+                                        true)
+                        .Cast<AssociativeNode>();
+
+                    var functionNodes = t.ClassNode.Procedures
+                        .OfType<FunctionDefinitionNode>()
+                        .Where(f => !f.MethodAttributes.HiddenInLibrary &&
+                                    !f.MethodAttributes.IsObsolete &&
+                                    f.Name != "_Dispose")
```

Comment:
@aparajit-pratap are there other hidden methods on imported classes that we may need to hide (ie don't show up in the node library)?

Reply:
These are all the filters that are applied to a `ProtoCore.DSASM.ProcedureNode`:
```
!x.MethodAttribute.HiddenInLibrary 
                            && !x.IsAssocOperator
                            && !x.IsAutoGenerated
                            && !x.Name.StartsWith(ProtoCore.DSASM.Constants.kInternalNamePrefix)
                            && !x.Name.Equals("Break")
                            && !CoreUtils.StartsWithDoubleUnderscores(x.Name);
```